staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go

```diff
@@ -133,24 +133,34 @@ func (c *codec) Decode(data []byte, defaultGVK *schema.GroupVersionKind, into ru
 		}
 	}
 
-	var strictDecodingErr error
+	var strictDecodingErrs []error
 	obj, gvk, err := c.decoder.Decode(data, defaultGVK, decodeInto)
 	if err != nil {
-		if obj != nil && runtime.IsStrictDecodingError(err) {
-			// save the strictDecodingError and the caller decide what to do with it
-			strictDecodingErr = err
+		if strictErr, ok := runtime.AsStrictDecodingError(err); obj != nil && ok {
+			// save the strictDecodingError and let the caller decide what to do with it
+			strictDecodingErrs = append(strictDecodingErrs, strictErr.Errors()...)
 		} else {
 			return nil, gvk, err
 		}
 	}
 
-	// TODO: look into strict handling of nested object decoding
 	if d, ok := obj.(runtime.NestedObjectDecoder); ok {
 		if err := d.DecodeNestedObjects(runtime.WithoutVersionDecoder{c.decoder}); err != nil {
-			return nil, gvk, err
+			if strictErr, ok := runtime.AsStrictDecodingError(err); obj != nil && ok {
```

Comment:
obj isn't returned from DecodeNestedObjects, checking it for nil here isn't relevant, right?

```suggestion
			if strictErr, ok := runtime.AsStrictDecodingError(err); ok {
```